### ISSUE TITLE
Proposal: option for disable multiple properties per line if braces not same line

### ISSUE THREAD

**Issue by evilebottnawi** [eslint/eslint#7722]
**Tell us about your environment**

* **ESLint Version:** 3.11.1
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```json
{
    "rules": {
        "object-property-newline": ["error", { "allowMultiplePropertiesPerLine": true }]
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
var obj = { foo: "foo", bar: "bar", baz: "baz" };

var obj2 = {
    foo: "foo", bar: "bar", baz: "baz"
};
```

**What did you expect to happen?**

Throws error in second example. If braces not on same line.

**What actually happened? Please include the actual, raw output from ESLint.**

Nothing

**Comment by not-an-aardvark** [comment]
Does [`object-curly-newline`](http://eslint.org/docs/rules/object-curly-newline) allow you to lint for this?

**Comment by evilebottnawi** [comment]
@not-an-aardvark nope, i'm talking about the newlines between properties.

**Comment by evilebottnawi** [comment]
I would like to allow writing a property only if the braces on the same line, but require a transition lines if braces not on same line.

**Comment by not-an-aardvark** [comment]
I see. So if I'm understanding correctly, you want to allow single-line objects if the entire object is on one line:

```js
({ a: 1, b: 2, c: 3 });
```

But if the object expression takes up multiple lines, you want every property to go on a new line:

```js
({
  a: 1,
  b: 2,
  c: 3
});
```

Is that correct?

**Comment by evilebottnawi** [comment]
@not-an-aardvark yes

**Comment by evilebottnawi** [comment]
@not-an-aardvark in conjunction with the https://github.com/eslint/eslint/issues/6488

**Comment by not-an-aardvark** [comment]
I see, thanks for clarifying.

I agree that it would be nice to have a way to enforce this. However, we should probably make sure that we don't introduce a conflict with the `object-curly-newline` rule by adding this option.

**Comment by evilebottnawi** [comment]
I do not think that conflict is possible:
`object-curly-newline` - responds for newlines after/before brackets
`object-property-newline` - responds for newlines between properties.

**Comment by not-an-aardvark** [comment]
Right, but even though this option would be enforcing newlines between properties, it would depend on whether newlines were present before/after brackets. I'm not saying there's definitely a conflict, it just seems like something that we should be on the lookout for.

**Comment by not-an-aardvark** [comment]
Closing this in favor of https://github.com/eslint/eslint/issues/7722. I think https://github.com/eslint/eslint/issues/7722 covers what this proposal is suggesting.

**Comment by piotr-s-brainhub** [comment]
@not-an-aardvark what do you mean `Closing this in favor of #7722. I think #7722 covers what this proposal is suggesting.`?

This is #7722
